docs/source/parameters.rst

```diff
@@ -0,0 +1,228 @@
+
+EXAMPLES:
+  find another sampling algorithm: latin hypercube, or something else from scikit-learn? what about stats models?
+
+What about adding reference of env block in pgen? (not modifying, just referencing)
+
+
+First, lets use the excellent built in package itertools to generate the parameters in the lulesh example specification:
+
+.. code-block:: python
+   :name: itertools_pgen.py
+   :caption: itertools_pgen.py
+   :linenos:
+
+   from maestrowf.datastructures.core import ParameterGenerator
+   import itertools as iter
+   
+   def get_custom_generator(env, **kwargs):
+       p_gen = ParameterGenerator()
+
+       sizes = (10, 20, 30)
+       iterations = (10, 20, 30)
+
+       size_values = []
+       iteration_values = []
+       trial_values = []
+       
+       for trial, param_combo in enumerate(iter.product(sizes, iterations)):
+           size_values.append(param_combo[0])
+           iteration_values.append(param_combo[1])
+           trial_values.append(trial)
+       
+       params = {
+           "TRIAL": {
+               "values": trial_values,
+               "label": "TRIAL.%%"
+           },       
+           "SIZE": {
+               "values": size_values,
+               "label": "SIZE.%%"
+           },
+           "ITER": {
+               "values": iteration_values,
+               "label": "ITER.%%"
+           },           
+       }
+
+       for key, value in params.items():
+           p_gen.add_parameter(key, value["values"], value["label"])
+    
+       return p_gen      
+
+This results in the following set of parameters, matching the lulesh sample workflow:
+
+.. table:: Sample parameters from itertools_pgen.py
+
+   =========== ==== ==== ==== ==== ==== ==== ==== ==== ====
+    Parameter   Values
+   ----------- --------------------------------------------
+    TRIAL        0    1    2    3    4    5    6    7    8
+   ----------- ---- ---- ---- ---- ---- ---- ---- ---- ----
+    SIZE        10   10   10   20   20   20   30   30   30
+   ----------- ---- ---- ---- ---- ---- ---- ---- ---- ----
+    ITER        10   20   30   10   20   30   10   20   30
+   =========== ==== ==== ==== ==== ==== ==== ==== ==== ====
+
+Pgen Arguments
+**************
+
+There is an additional pgen feature that can be used to make them more dynamic.  The above example generates a fixed set of parameters, requiring editing the itertools_pgen.py file to change that.  Maestro supports passing arguments to these generator functions on the command line:
+
+
+.. code-block:: bash
+
+   $ maestro run study.yaml --pgen itertools_pgen_pargs.py --parg "SIZE_MIN:10" --parg "SIZE_STEP: 10" --parg "NUM_SIZES:4"
```

Comment:
There's a space in one of the args, I think the space won't be stripped out.

Reply:
whoops, good catch